packaging/dependencies.nix

```diff
@@ -61,6 +61,14 @@ let
       workDir = null;
     };
 
+  # Work around weird `--as-needed` linker behavior with BSD, see
+  # https://github.com/mesonbuild/meson/issues/3593
+  bsdNoLinkAsNeeded = finalAttrs: prevAttrs: lib.optionalAttrs stdenv.hostPlatform.isBSD {
+    postConfigure = prevAttrs.postConfigure or "" + ''
+       sed -i build.ninja -e 's/-Wl,--as-needed //'
+    '';
+  };
```

Comment:
Please instead use the meson option `-D b_asneeded=false` to disable generating this option at all.

Reply:
Good idea, switched it!